### DISCUSSION TITLE
Allow request with query param, but disallow "find" on collection

### DISCUSSION THREAD

**Discussion by ruslan-mystore** [strapi/strapi#7958] [Legacy Topics]
While making a request to the REST API, I want  to show a filtered result with query param
https://api.example.com/users?uid=dl982kdja

Strapi providing this when the collection's (e.g users) permission "find" set to true.

**What is the expected behavior?**
How to make a request such as https://api.example.com/users?uid=dl982kdja
So the collection users permission "find" set to false. I just don't want to expose 10,000 users 

Is there any workaround for that ?

**Comment by creotip** [comment]
Here is what I did.
Let's say you have restaurant collection. 
Go to /api/restaurant/services/restaurant.js

Replace with the following code:
```
'use strict';


/**
 * Read the documentation (https://strapi.io/documentation/v3.x/concepts/services.html#core-services)
 * to customize this service
 */

module.exports = {
    find(params, populate) {
        
        if ("uid" in params) {
           return strapi.query('gift').find(params, populate);
        }
        return 'no content'
      },
};


```

**Reply by ruslan-mystore** [reply]
great solution.... thanks
